### ISSUE TITLE
Decouple --verbose from --log-level

### ISSUE THREAD

**Issue by felipesanches** [googlefonts/fontbakery#2693]
@yanone brought up a valid concern at https://github.com/googlefonts/fontbakery/issues/2690#issuecomment-553303724

We should undo the equivalency we have had for a long time between `--verbose` and `--log-level PASS`. The `--verbose` flag should have its own independent meaning for the sake of clarity. Determining verbosity and selecting log-level are similar concepts but not exactly the same thing and some users such as @yanone may have a use case requiring independent setup of both options.

**Comment by graphicore** [comment]
So, should we move the discussion over here? Or discuss there and then put our conclusion in here?

**Comment by felipesanches** [comment]
That issue was about something else and was already closed. Let's continue the conversation here.

**Comment by graphicore** [comment]
But the context is there.

**Comment by felipesanches** [comment]
we have hyperlinks in place for people to read the related stuff when needed ;-)

**Comment by felipesanches** [comment]
but feel free to post snippets of conversation from there when replying here

**Comment by yanone** [comment]
My context:
A font I was working on had a longer list of missing glyphs which were mentioned in a WARNING. The list of missing glyphs in the output were cropped and there was no way for me to see the entire list. Felipe said that some people want to see less output, while I *need* to see the *entire* output for this warning, or otherwise I can't fix the font. I literally had to calculate the missing glyphs myself, which is stupid because they were already calculated, but partially hidden from me.
Instead, Felipe offered to print the entire list of missing glyphs in an INFO message.

Here's my case:
That people sometimes want to see sometimes less and sometimes more output is a clear case for a verbosity level switch. Each message would then print less or more output. For example, the rationale doesn't need to be printed for a low verbosity level.
The verbosity level is independent of the log level of each message, as the log level describes the category of a message. I'm mostly interested in reading warnings, as the requirement of my work is to submit fonts lacking warnings, really. And I don't care about info messages, as they are the least important ones.
Now, printing the list of missing glyphs in an info message means putting the output in a different place than it belongs. It really belongs to the exact same warning message, because the missing glyphs constitute a real failure of this font, and that warning message is where I want to see the output.
Log levels (=message types) and verbosity are two separate concepts.

**Comment by graphicore** [comment]
Ok, back on topic.

I think the statement by @yanone in https://github.com/googlefonts/fontbakery/issues/2690#issuecomment-553303724:

> you're mixing up log level and verbosity

is wrong, because it is an anachronism. The current interpretation came first.

Hence I wrote in https://github.com/googlefonts/fontbakery/issues/2690#issuecomment-553381585:

> No, you are separating it.

As you are separating log level and verbosity where no separation is intended.

I wrote in https://github.com/googlefonts/fontbakery/issues/2690#issuecomment-553381585:

> [...] the intention of having many different log entries with different log levels in one check output is to be able to do exactly that.  

I think we are fine with the status quo and adding another mechanism to show more or less information is just making things more complicated, but does not add value because we can already do that. I'm being practical here. Verbosity and log levels are coupled in Font Bakery: log levels are the way to transparently express importance of the message, verbosity level is to select the output by importance.

> The verbosity level is independent of the log level of each message, as the log level describes the category of a message.

This adds a new interpretation to what we're doing right now and it is more complicated. I see where you are going here, but I don't think this kind of abstraction, between "category" and "verbosity level" is necessary.

> I'm mostly interested in reading failures, as the requirement of my work is to submit fonts lacking failures, really. And I don't care about info messages, as they are the least important ones.

You shouldn't care about the "category" or "message type" (your concept) but about the information and how to get to the information you need. If that means caring about also showing the INFO messages of a check I don't see a problem.

> Now, printing the list of missing glyphs in an info message means putting the output in a different place than it belongs. It really belongs to the exact same failure message, because the missing glyphs constitute a real failure of this font, and that failure message is where I want to see the output.

The output is not in a different place, it is enclosed within the context of the specific check execution. The messages of one check are guaranteed to be in order, we guarantee that that INFO is directly after or before (wherever the author puts it) that FAIL message. It's in fact in the same place as far as the possibilities of linear output go.

All you do is to suggest an even more compartmentalized messaging system and argue that it is not enough to have a checks output partitioned into separate levels of importance. *The granularity of controlling the output should go into the message itself.*

>  Log levels (=message types) and verbosity are two separate concepts.

No, they are not, we could make them separate, but they *are* not.

> So what I need for my own case is to print only failures, but in their highest verbosity.

To do this you would print checks with a status of FAIL and with a message log level of INFO (or below).

```
$ fontbakery check-googlefonts -l FAIL -m INFO /your/fonts/*.ttf
```

> For example, the rationale doesn't need to be printed for a low verbosity level.

I actually agree with this. I'd rather like to see it printed alongside with e.g. `-m INFO`.

**Comment by felipesanches** [comment]
Now I feel uncertain about it all. I can see how confusing and frustrating the current approach can be to users (like people with a use case similar to @yanone's). But I also understand the concerns brought up by @graphicore.

**Comment by laerm0** [comment]
Maybe a switch that flips between "old mode" and "new mode" where the `--verbose` switch is possible?

**Comment by yanone** [comment]
Log levels and verbosity aren't concepts either of us invented. They are common in computer science and are therefore expected to behave in a certain way:
Log level: Categorize messages with their importance to be able to filter them
Verbosity: Describe a situation with different amounts of words. 

All this isn't important to me at all. I can work with it. But the problem might come up again in the future for other use cases, and Felipe said "rethink", so I wanted to point out that at the very least you're using the term "verbosity" wrong.

**Comment by graphicore** [comment]
I disagree slightly. Yes, we didn't invent those terms. No, there's no general agreement about what they mean and how they are implemented.
